packages/pluggableWidgets/image-native/src/components/ImageComponents.tsx

```diff
@@ -0,0 +1,219 @@
+                name={name}
+                source={source}
+                customWidth={customWidth}
+                customHeight={customHeight}
+                iconSize={iconSize}
+                initialDimensions={initialDimensions}
+                onClick={onClickType === "enlarge" ? () => setEnlarged(true) : onClick}
+                styles={styles}
+            />
+            <GetImageDimensionsComponent
+                name={name}
+                source={source}
+                initialDimensions={initialDimensions}
+                onLayoutSetInitialDimensions={onLayoutSetInitialDimensions}
+            />
+            <ImageEnlarged
+                name={name}
+                visible={enlarged}
+                setEnlarged={setEnlarged}
+                source={source}
+                initialDimensions={initialDimensions}
+                styles={styles}
+            />
+        </Fragment>
+    );
+};
+
+export const GetImageDimensionsComponent: FunctionComponent<GetImageDimensionsComponentProps> = props => {
+    const { source, initialDimensions, onLayoutSetInitialDimensions, name } = props;
+
+    /* Render dynamicSVG once to get initial dimensions */
+    return source?.image &&
+        source.type === "dynamicSVG" &&
+        (!initialDimensions?.width || !initialDimensions?.height) ? (
+        <View
+            style={{
+                position: "absolute",
+                top: 0,
+                left: 0,
+                width: "100%",
+                height: "100%",
+                opacity: 0
+            }}
+            pointerEvents={"none"}
+        >
+            <SvgUri
+                testID={`${name}$SvgUriTemporary`}
+                uri={source.image as string}
+                onLayout={onLayoutSetInitialDimensions}
+                style={{
+                    opacity: 0
+                }}
+            />
+        </View>
+    ) : null;
+};
+
+export const ImageSmall: FunctionComponent<ImageSmallProps> = props => {
+    const { source, initialDimensions, customWidth, customHeight, iconSize, onClick, styles, name } = props;
+    const [dimensions, setDimensions] = useState<DimensionsType>();
+    const [svgProps] = extractStyles(styles.image as ImageStyle, ["width", "height"]);
+    const onLayoutSetDimensionsCallback = useCallback(
+        ({ nativeEvent: { layout } }: LayoutChangeEvent) => {
+            onLayoutSetDimensions(
+                customWidth ?? layout.width,
+                customHeight ?? layout.height,
+                setDimensions,
+                initialDimensions
+            );
+        },
+        [initialDimensions, customWidth, customHeight]
+    );
+
+    return source.type === "icon" || (initialDimensions?.width && initialDimensions?.height) ? (
+        <Pressable
+            testID={`${name}$ImageSmallPressable`}
+            onPress={onClick}
+            onLayout={
+                source.type !== "icon" && (!dimensions?.width || !dimensions?.height)
```

Comment:
this check `source.type !== "icon" && (!dimensions?.width || !dimensions?.height)` is here and line 151. This could be a var

Reply:
Done